### PR TITLE
Add customFieldValue method to Context

### PR DESCRIPTION
**Pull request by calthejuggler**
This PR adds the customFieldValue method to the `Context` class which gets a particular custom field value and parses it into it's appropriate type.

### CODE REVIEW

**Comment by calthejuggler** [comment]
Current dependencies on/for this PR:
* main
  * **PR #34** <a href="https://app.graphite.dev/github/pr/absmartly/javascript-sdk/34" target="_blank">Graphite</a> 
    * **PR #35** <a href="https://app.graphite.dev/github/pr/absmartly/javascript-sdk/35" target="_blank">Graphite</a>  👈
      * **PR #36** <a href="https://app.graphite.dev/github/pr/absmartly/javascript-sdk/36" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/absmartly/javascript-sdk/35?utm_source=stack-comment).

**Review by mario-silva — APPROVED** [review]

File: src/context.ts [hunk 1]
```diff
@@ -634,6 +634,45 @@ export default class Context {
...
+			if (field != null) {
+				switch (field.type) {
+					case "text":
+					case "string":
+						return field.value;
+					case "number":
+						return Number(field.value);
+					case "json":
+						try {
+							return JSON.parse(field.value);
```
Comment (marcio-absmartly) [inline src/context.ts:651]: I think JSON.parse will fail for empty strings and null. Maybe we shouldn't error in that case?

File: src/context.ts [hunk 2]
```diff
@@ -634,6 +634,45 @@ export default class Context {
...
+						} catch (e) {
+							console.error(`Failed to parse JSON custom field value '${key}' for experiment '${experimentName}'`);
+							return null;
+						}
+					case "boolean":
+						return field.value === "true";
+					default:
+						console.error(
+							`Unknown custom field type '${field.type}' for experiment '${experimentName}' and key '${key}'`
```
Comment (marcio-absmartly) [inline src/context.ts:660]: Here we could add in the message "You may need to upgrade SDK to latest version"

**Review by marcio-absmartly — CHANGES_REQUESTED** [review]
See comments

File: src/context.ts (hunk 1 above)
Comment (calthejuggler) [reply src/context.ts:651]: Null seems to be okay, but strings aren't. I'll change it :)
image

File: src/context.ts [hunk 3]
```diff
@@ -634,6 +634,47 @@ export default class Context {
...
+					case "text":
+					case "string":
+						return field.value;
+					case "number":
+						return Number(field.value);
+					case "json":
+						try {
+							return JSON.parse(field.value);
+						} catch (e) {
+							if (field.value === "null") return null;
```
Comment (marcio-absmartly) [inline src/context.ts:653]: I think this test should be done before trying to parse the JSON.

Comment (calthejuggler) [reply src/context.ts:651]: Updated :)

**Review by marcio-absmartly — APPROVED** [review]

**Pull request merged**